packages/vite/src/node/config.ts

```diff
@@ -402,69 +413,88 @@ export async function resolveConfig(
   }
 
   // TODO Deprecation warnings - remove when out of beta
-  if (config.build?.base) {
+
+  const logDeprecationWarning = (
```

Comment:
**question:** Why are we updating code for deprecation?
Also what is this `beta`? From when is it?

Reply:
Ah okay... seeing now that you created this to solve all the duplication below
But what has it to do with the goal of the PR?
Do we need to change the git blame annotations for that in this PR?
IMO it's more worth about thinking to remove this if it's for deprecation :thinking:

Reply:
Not sure about when these were intended to be removed - but since they're still part of the API, we couldn't remove them entirely in a minor version.

I think it's okay to do a bit of refactoring like this in a PR, but don't have too strong an opinion here - I can remove it if you like.

Reply:
I will let @patak-js decide :slightly_smiling_face:

Reply:
=> https://github.com/vitejs/vite/pull/2991#discussion_r624555056

Reply:
Haha ^^ didn't read the discussions from you and patak before
But seems we had kinda same thoughts :sweat_smile:

Reply:
I also prefer if they are separated PRs, mainly because this is could be an interesting PR to reference later. But the refactoring is contained. If @antfu reviews the PR and asks the same question, it is a clear indication that it better to separate them.

Reply:
PRs are cheap, here you go: #3232